### ISSUE TITLE
bignum literals aren't interpreted correctly

### ISSUE THREAD

**Issue by GBeauregard** [microsoft/pyright#2860]
**Describe the bug**
Pyright mishandles the value of large literals.

**To Reproduce**

```python
a = 90000000000000001
reveal_type(a)  # Literal[90000000000000000]
```

**Expected behavior**
`Literal[90000000000000001]`, or maybe bailing to `int`?

**VS Code extension or command-line**
pyright 1.1.208 Linux commandline

**Additional context**
I thought about this while [exploring ways literal math can break](https://github.com/python/mypy/issues/11990#issuecomment-1013328832). This is highly pathological, but I was thinking this *may* be more relevant with the introduction of literal math. Should this bail to `int` or be left as-is? The alternative is to implement bignum in pyright, but that seems a bit much for something so pathological.

The implication is the false negative:
```python
from typing import Literal
b: Literal[90000000000000000] = 90000000000000001  # no error
```
mypy does catch the error (presumably because it's coded in python which is bignum by default)

**Comment by erictraut** [comment]
Since this is just a theoretical issue, I'm going to ignore it for now. I don't see a straightforward way to address it. If it becomes an actual problem, I am willing to reopen the issue and invest in a solution.
